core/templates/dev/head/pages/community-dashboard-page/community-dashboard-page.controller.ts

```diff
@@ -87,13 +88,50 @@ angular.module('oppia').directive('communityDashboardPage', [
           ctrl.$onInit = function() {
             ctrl.profilePictureDataUrl = null;
             ctrl.username = null;
+            ctrl.userIsReviewer = false;
+            ctrl.userCanReviewTranslationSuggestionsInLanguages = [];
+            ctrl.userCanReviewVoiceoverSuggestionsInLanguages = [];
+            ctrl.userCanReviewQuestions = false;
 
             UserService.getProfileImageDataUrlAsync().then(function(dataUrl) {
               ctrl.profilePictureDataUrl = dataUrl;
             });
 
             UserService.getUserInfoAsync().then(function(userInfo) {
-              ctrl.username = userInfo.getUsername();
+              if (userInfo.isLoggedIn()) {
+                ctrl.username = userInfo.getUsername();
+                UserService.getUserCommunityRightsData().then(
+                  function(userCommunityRights) {
+                    userCommunityRights
+                      .can_review_translation_for_language_codes.forEach(
+                        function(languageCode) {
+                          ctrl.userCanReviewTranslationSuggestionsInLanguages
+                            .push(
+                              LanguageUtilService.getAudioLanguageDescription(
+                                languageCode));
+                        });
+
+                    userCommunityRights
+                      .can_review_voiceover_for_language_codes.forEach(function(
+                          languageCode) {
+                        ctrl.userCanReviewVoiceoverSuggestionsInLanguages.push(
+                          LanguageUtilService.getAudioLanguageDescription(
+                            languageCode));
+                      });
+
+                    ctrl.userCanReviewQuestions = (
+                      userCommunityRights.can_review_questions);
+
+                    if (
+                      ctrl.userCanReviewTranslationSuggestionsInLanguages
+                        .length > 0 ||
+                      ctrl.userCanReviewVoiceoverSuggestionsInLanguages
+                        .length > 0 ||
+                      ctrl.userCanReviewQuestions) {
```

Comment:
This conditional is difficult to read. Consider extracting this to a variable or method with an explicit name.